examples/tutorials/nb_python/material_tutorial.py

```diff
@@ -0,0 +1,198 @@
+# ---
+
+# %%
+# !curl -L https://raw.githubusercontent.com/facebookresearch/habitat-sim/master/examples/colab_utils/colab_install.sh | NIGHTLY=true bash -s
+
+# %%
+# %cd /content/habitat-sim
+import math
+import os
+import random
+import sys
+
+import git
+import magnum as mn
+import numpy as np
+from matplotlib import pyplot as plt
+
+import habitat_sim
+from habitat_sim.gfx import PhongMaterialInfo
+from habitat_sim.utils.common import quat_from_angle_axis
+
+if "google.colab" in sys.modules:
+    os.environ["IMAGEIO_FFMPEG_EXE"] = "/usr/bin/ffmpeg"
+
+repo = git.Repo(".", search_parent_directories=True)
+dir_path = repo.working_tree_dir
+data_path = os.path.join(dir_path, "data")
+
+
+def place_agent(sim):
+    # place our agent in the scene
+    agent_state = habitat_sim.AgentState()
+    agent_state.position = [5.0, 0.0, 1.0]
+    agent_state.rotation = quat_from_angle_axis(
+        math.radians(70), np.array([0, 1.0, 0])
+    ) * quat_from_angle_axis(math.radians(-20), np.array([1.0, 0, 0]))
+    agent = sim.initialize_agent(0, agent_state)
+    return agent.scene_node.transformation_matrix()
+
+
+def make_configuration():
+    # simulator configuration
+    backend_cfg = habitat_sim.SimulatorConfiguration()
+    backend_cfg.scene.id = "data/scene_datasets/habitat-test-scenes/van-gogh-room.glb"
+    backend_cfg.enable_physics = True
+
+    # agent configuration
+    sensor_cfg = habitat_sim.SensorSpec()
+    sensor_cfg.resolution = [1080, 960]
+    agent_cfg = habitat_sim.agent.AgentConfiguration()
+    agent_cfg.sensor_specifications = [sensor_cfg]
+
+    return habitat_sim.Configuration(backend_cfg, [agent_cfg])
+
+
+def show_obs(sim):
+    data = sim.get_sensor_observations()["rgba_camera"]
+    plt.figure(figsize=(12, 12))
+    plt.imshow(data, interpolation="nearest")
+    plt.axis("off")
+    plt.show(block=False)
+    plt.pause(1)
+
+
+if __name__ == "__main__":
+
+# %%
+    # set up the scene and agent (camera position)
+
+    cfg = make_configuration()
+    sim = habitat_sim.Simulator(cfg)
+    place_agent(sim)
+
+# %%
+    # place a sphere in the scene
+    obj_templates_mgr = sim.get_object_template_manager()
+
+    sphere_template_id = obj_templates_mgr.load_object_configs(
+        str(os.path.join(data_path, "test_assets/objects/sphere"))
+    )[0]
+
+    sphere_ids = []
+    sphere_ids.append(sim.add_object(sphere_template_id))
+    sim.set_translation([3.7, 0.23, 0.0], sphere_ids[0])
+
+    show_obs(sim)
+
+# %%
+    # change the material for the sphere render asset
+
+    # could also use sim.get_object_template(sphere_template_id) instead of
+    # get_object_initialization_template
+    render_asset_handle = sim.get_object_initialization_template(
+        sphere_ids[0]
+    ).render_asset_handle
+    num_materials = sim.get_num_render_asset_materials(render_asset_handle)
+    assert num_materials == 1
+    material_index = 0
+
+    material = sim.get_render_asset_material(render_asset_handle, material_index)
```

Comment:
A concern from the earlier PR: I'm not exposing our C++ material IDs to python. Instead, the python can only refer to a material by `(render_asset_handle, material_index)`. This will be a problem if we eventually get materials from other sources, such as created from scratch or loaded from something other than a render asset. More broadly, we'd like a consistent way of referencing assets in general (materials, meshes, textures, other things?) at the python level.

Reply:
After discussion with the infra team, I'm punting for now. I've mostly left the API as-is. I've removed the python API for overriding a material for an object because that was probably the strangest API here and most likely to change. I'm open to changes here before merging this PR.